### PR TITLE
Implement BEGAN

### PR DESCRIPTION
**Pull request by radujica**
Also allows kwargs in _end_iteration for custom printing

### CODE REVIEW

File: vegans/began.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                """ Update weights
+                """
+                diff = gamma * loss_D_real.item() - loss_D_fake.item()
+
+                # Update weight term
+                k = k + lambda_k * diff
+                k = min(max(k, 0), 1)  # Constraint to interval [0, 1]
+
+                # Update convergence metric
+                # M = (loss_D_real + torch.abs(diff)).data
```
Comment (hrzn) [inline vegans/began.py]: Would be nice to display/store this. You can remove the line in the meantime.

File: vegans/gan.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def _default_optimizers(self,):
         optimizer_G = optim.Adam(self.generator.parameters(), lr=0.0002, betas=(0.5, 0.999))
         return optimizer_D, optimizer_G
 
-    def _end_iteration(self, epoch, minibatch_iter, G_loss=None, D_loss=None, D_x=None, D_G_z1=None, D_G_z2=None):
+    def _end_iteration(self, epoch, minibatch_iter, G_loss=None, D_loss=None, **kwargs):
```
Comment (hrzn) [inline vegans/gan.py]: Much better indeed ;)

File: vegans/began.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                real = data.to(self.device)
+                batch_size = real.size(0)
+
+                """ Train the generator
+                """
+                self.optimizer_G.zero_grad()
+
+                # generate fake
+                noise = torch.randn(batch_size, self.nz, device=self.device)
+                fake = self.generator(noise).detach()
```
Comment (hrzn) [inline vegans/began.py]: Are you sure you want to `detach()` here? It seems this would prevent training the generator.

Comment (radujica) [reply vegans/began.py]: great catch! :) fixed it
